File: Last_Shot.c
#include<stdio.h>
#include<string.h> 
#include <stdlib.h>

// char* transtime(char *time,int x);


int main(){
    FILE *fp1,*fp2;
    char buf[50];
    char tmp[40];
    char time_in[6];
    char describe[40];
    char out[6];
    char minutes[6];
    char seconds[6];
    int m,s,t,x;
    printf("สฃำเร๋สý:\n");
    scanf("%d",&x);
    fp1=fopen("timeline.txt","r");
    fp2=fopen("timeline_out.txt","w");
    if(!fp1){
        printf("can't open file1\n");
    }
    if(!fp2){
        printf("can't open file2\n");
    }
    while(fgets(buf, sizeof(buf), fp1))
    {
        strcpy(time_in,strtok(buf," "));
        strcpy(describe,strtok(NULL," "));
        // printf("%s\n",time_in,describe);

        strcpy(minutes,strtok(time_in,":"));
        strcpy(seconds,strtok(NULL,":"));
        m=atoi(minutes);
        s=atoi(seconds);
        // printf("%d %d\n",m,s);

        memset(minutes,0,sizeof(minutes)/sizeof(char));
        memset(seconds,0,sizeof(seconds)/sizeof(char));
        t=(m*60+s)-(90-x);
        m=t/60;
        s=t%60;
        itoa(m,minutes,10);
        strcat(minutes,":");
        itoa(s,seconds,10);
        // printf("%s %s\n",minutes,seconds);
        if(s<0)break;
        strcpy(out,strcat(minutes, seconds));
        printf("%s %s",out,describe);
        fprintf(fp2,"%s %s",out,describe);
    }
    fclose(fp1);
    fclose(fp2);
    return 0;
}

File: Last_Shot.py
import sys

with open("config.txt", "r", encoding="utf-8") as f1:
    t = int(f1.readline().strip("剩余时间(秒):"))

with open("timeline.txt", "r", encoding="utf-8") as f2:
    with open("timeline_out.txt", "w", encoding="utf-8") as f3:
        f3.write("剩余时间为"+str(t)+"秒\n")
        minutes=int(t/60)
        seconds=t%60
        if seconds<10:
            seconds='0'+str(seconds)
        f3.write(str(minutes)+':'+str(seconds)+" 战斗开始\n")
        line = f2.readlines()
        for i in range(len(line)):
            line[i] = line[i].replace('　',' ')
            line[i] = line[i].replace('\n',' ')
            line[i]=line[i].split(" ")
            if line[i][0][0].isdigit()==False:
                for j in range(len(line[i])):
                    f3.write(line[i][j]+" ")
                f3.write('\n')
                continue
            if line[i][0].find(':', 0, len(line[i][0]))>0:
                m,s=line[i][0].strip().split(':')
                time=(int(m)*60+int(s))-(90-t)
                m=int(time/60)
                s=time%60
                if time<0:
                    sys.exit(0)
                if(s<10):
                    s='0'+str(s)
                line[i][0]=str(m)+':'+str(s)
            else:
                if len(line[i][0])==3:
                    time=(int(line[i][0][0])*60+int(line[i][0][1])*10+int(line[i][0][2]))-(90-t)
                else:
                    time=int(line[i][0])-(90-t)
                    if time<0:
                        sys.exit(0)
                if time>60:
                    if time-60<10:
                        time="0"+str(time-60)
                    else:
                        time=str(time-60)
                    time="1"+time
                line[i][0]=str(time)
            for j in range(len(line[i])):
                f3.write(line[i][j]+" ")
            f3.write('\n')